### PR TITLE
Feat/add dark mode

### LINKED ISSUE #48: [Feat] - Dark theme

### PR DESCRIPTION
**Pull request by sanchis**
fixes #48 

- [X] Move project from css to scss to easy change dark and light mode.
- [x] Create a service to switch between styles.

### CODE REVIEW

File: client/src/app/services/theme.service.ts.service.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
```
Comment (Darkweak) [inline client/src/app/services/theme.service.ts.service.ts:1]: Rename this file to `theme.service.ts`

File: client/src/app/app.component.ts [hunk 2]
```diff
@@ -1,19 +1,24 @@
-import { Component } from '@angular/core';
+import { Component, HostBinding } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Router } from '@angular/router';
 import { DecodedToken, Token } from 'src/app/storage/Token';
+import { MatDialog } from '@angular/material/dialog';
+import { Theme, ThemeService } from './services/theme.service.ts.service';
```
Comment (Darkweak) [inline client/src/app/app.component.ts:6]: Import from `src`

**Review by Darkweak — CHANGES_REQUESTED** [review]
Thanks for contributing @sanchis. 
IMO it would be awesome if you can change the button for a switch or a changing icon (like sun and moon for example).

File: client/src/app/app.component.ts (hunk 2 above)
Comment (sanchis) [reply client/src/app/app.component.ts]: Fixed in: https://github.com/CMEPW/Smersh/pull/49/commits/69d0cef7dab090e012dcf0784c237b7c1797cb18

File: client/src/app/services/theme.service.ts.service.ts (hunk 1 above)
Comment (sanchis) [reply client/src/app/services/theme.service.ts.service.ts]: Fixed in: https://github.com/CMEPW/Smersh/pull/49/commits/69d0cef7dab090e012dcf0784c237b7c1797cb18

File: client/src/app/components/side-bar/side-bar.component.html [hunk 3]
```diff
@@ -4,6 +4,16 @@
...
+        <button mat-raised-button  (click)="themeService.toggleTheme()">
+            <span [ngSwitch]="themeService.currentTheme">
+                <span *ngSwitchCase="'dark-theme'">
+                    Light theme
+                </span>
+                <span *ngSwitchDefault>
+                    Dark theme
+                </span>
+            </span>
+        </button>
```
[the remark this answers is not in the document]
Comment (sanchis) [reply client/src/app/components/side-bar/side-bar.component.html]: Fixed on: https://github.com/CMEPW/Smersh/pull/49/commits/a912ad03bd3eaf3f8db36ddef60e7f09bc085b4f
Preview
test

[the remark this answers is not in the document]
Comment (Darkweak) [reply client/src/app/components/side-bar/side-bar.component.html]: Really nice (cc @jenaye)

**Review by Darkweak — APPROVED** [review]
LGTM ✅

**Comment by jenaye** [comment]
Great job @sanchis thx a lot 🔥
I added you to contributor list in readme

**Comment by sanchis** [comment]
> Great job @sanchis thx a lot 🔥
> I added you to contributor list in readme

Thanks!! 🤘

**Pull request merged**